File: POO/TP2/Practica Python/EJ10/Persona.py
class Persona:
    __edad: int
    __sexo: str
    __estudia: bool
    __trabaja: bool
    _puede_trabajar=False
    _puede_manejar=False
    
    def __init__(self, edad, sexo, estudia, trabaja):
        self.__edad= edad
        if edad >= 16:
            self._puede_trabajar = True
            if edad>=17:
                self._puede_manejar= True
        self.__sexo= sexo
        self.__estudia= estudia
        self.__trabaja= trabaja
    
    def get_edad(self):
        return self.__edad
    
    def set_edad(self,edad):
        self.__edad=edad
        
    def get_sexo(self):
        return self.__sexo
    
    def set_sexo(self,sexo):
        self.__sexo=sexo
        
    def get_estudia(self):
        return self.__estudia
    def set_estudia(self,estudia):
        self.__estudia=estudia
    
    def get_trabaja(self):
        return self.__trabaja
    def set_trabaja(self,trabaja):
        self.__trabaja=trabaja
    
    def get_puede_trabajar(self):
        return self._puede_trabajar
    
    def get_puede_manejar(self):
        return self._puede_manejar

File: POO/TP2/Practica Java/EJ10/Persona.java
package EJ10;

public class Persona {
    private Integer edad;
    private String sexo;
    private boolean estudia;
    private boolean trabaja;
    private boolean puede_trabajar = false;
    private boolean puede_manejar = false;

    public Persona(Integer edad, String sexo, boolean estudia, boolean trabaja) {
        this.edad = edad;
        this.sexo = sexo;
        this.estudia = estudia;
        this.trabaja = trabaja;
        this.permiso();
    }

    private void permiso() {
        if (this.edad >= 16) {
            this.puede_trabajar = true;
            if (this.edad >= 17) {
                this.puede_manejar = true;
            }
        }
    }

    public boolean getPuede_trabajar() {
        return this.puede_trabajar;
    }

    public void setPuede_trabajar(boolean puede_trabajar) {
        this.puede_trabajar = puede_trabajar;
    }

    public boolean getPuede_manejar() {
        return this.puede_manejar;
    }

    public void setPuede_manejar(boolean puede_manejar) {
        this.puede_manejar = puede_manejar;
    }

    public Integer getEdad() {
        return this.edad;
    }

    public void setEdad(Integer edad) {
        this.edad = edad;
    }

    public String getSexo() {
        return this.sexo;
    }

    public void setSexo(String sexo) {
        this.sexo = sexo;
    }

    public boolean getEstudia() {
        return this.estudia;
    }

    public void setEstudia(boolean estudia) {
        this.estudia = estudia;
    }

    public boolean getTrabaja() {
        return this.trabaja;
    }

    public void setTrabaja(boolean trabaja) {
        this.trabaja = trabaja;
    }

}
